index.html

```diff
@@ -55,213 +55,331 @@
-        algorithm. The algorithm takes an <a>ImageResource</a> <var>image</var>. This algorithm will return a set.
-      </p>
       <ol>
-        <li>Let <var>set</var> be an empty <a>set</a>.
-        </li>
-        <li>Let <var>value</var> be <var>image.sizes</var>.
-        </li>
-        <li>If <a>Type</a>(<var>value</var>) it not String, return
-          <code>undefined</code>.
+        <li>Let <var>imageResources</var> be a new Array object created as if
+        by the expression [].
         </li>
-        <li>Otherwise, parse <var>value</var> as if it was a [[HTML]]
-          {{HTMLLinkElement/sizes}} attribute and let <a>list</a>
-          <var>keywords</var> be the result.
+        <li>
+          <a>For each</a> <var>entry</var> of <var>entries</var>:
         </li>
         <li>
-          <a>For each</a> <var>keyword</var> of <var>keywords</var>:
           <ol>
-            <li>
-              <a data-lt="set-append">Append</a> <var>keyword</var>,
-              <a>lowercased</a>, to <var>set</var>.
+            <li>If <var>entry</var>["src"] is not <code>undefined</code>:
+              <ol>
+                <li>Let <var>image</var> be a new object created as if by the
+                expression ({}).
+                </li>
+                <li>Set <var>image</var>["src"] to the result of <a>parsing</a>
+                <var>entry</var>["src"] using <var>context</var> as the
+                <var>base URL</var>.
+                </li>
+                <li>Set <var>image</var>["type"] to the result of running
+                <a>processing the <code>type</code> member of an image</a>
+                given <var>entry</var>.
+                </li>
+                <li>Set <var>image</var>["sizes"] to the result of running <a>
+                  processing the <code>sizes</code> member of an image</a>
+                  given <var>entry</var>.
+                </li>
+                <li>
+                  <a data-lt="list-append">Append</a> <var>image</var> to
+                  <var>imageResources</var>
+                </li>
+              </ol>
             </li>
           </ol>
         </li>
-        <li>Return <var>set</var>.
+        <li>Return <var>imageResources</var>.
         </li>
       </ol>
+      <div class="note">
+        The choice of what to do with the Array&lt;<a>ImageResource</a>&gt;
+        returned from this algorithm is left to the implementation context in
+        which it is being used. Regardless of the implementation context,
+        however, the order of the <a>ImageResource</a> members provided MUST be
+        the order in which they are returned. This will enable each
+        implementation context to consider the order of the
+        <a>ImageResource</a> members provided by the author in their selection
+        algorithm, should they so choose.
+      </div>
     </section>
-    <section data-dfn-for="ImageResource" data-link-for="ImageResource">
+    <section class="appendix">
```

Comment:
Generally, appendix's are non-normative... so best not make this an appendix, as the ECMA stuff is likely normative. 

```suggestion
    <section>
```